fili-generic-example/src/main/java/com/yahoo/wiki/webservice/data/config/table/GenericTableLoader.java

```diff
@@ -0,0 +1,171 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.wiki.webservice.data.config.table;
+
+import com.yahoo.bard.webservice.data.config.ResourceDictionaries;
+import com.yahoo.bard.webservice.data.config.dimension.DimensionConfig;
+import com.yahoo.bard.webservice.data.config.names.ApiMetricName;
+import com.yahoo.bard.webservice.data.config.names.FieldName;
+import com.yahoo.bard.webservice.data.config.table.BaseTableLoader;
+import com.yahoo.bard.webservice.data.config.table.PhysicalTableDefinition;
+import com.yahoo.bard.webservice.data.time.TimeGrain;
+import com.yahoo.bard.webservice.data.time.ZonedTimeGrain;
+import com.yahoo.bard.webservice.data.time.ZonelessTimeGrain;
+import com.yahoo.bard.webservice.druid.model.query.AllGranularity;
+import com.yahoo.bard.webservice.druid.model.query.Granularity;
+import com.yahoo.bard.webservice.table.TableGroup;
+import com.yahoo.bard.webservice.util.Utils;
+import com.yahoo.wiki.webservice.data.config.auto.ConfigLoader;
+import com.yahoo.wiki.webservice.data.config.auto.DruidConfig;
+import com.yahoo.wiki.webservice.data.config.dimension.GenericDimensions;
+import com.yahoo.wiki.webservice.data.config.metric.DruidMetricName;
+import com.yahoo.wiki.webservice.data.config.metric.FiliMetricName;
+import com.yahoo.wiki.webservice.data.config.metric.MetricNameGenerator;
+
+import org.joda.time.DateTimeZone;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Load the Wikipedia-specific table configuration.
+ */
+public class GenericTableLoader extends BaseTableLoader {
+    private final Map<String, Set<Granularity>> validGrains =
+            new HashMap<>();
+    // Set up the metrics
+    private final Map<String, Set<FieldName>> druidMetricNames =
+            new HashMap<>();
+    private final Map<String, Set<ApiMetricName>> apiMetricNames =
+            new HashMap<>();
+    // Set up the table definitions
+    private final Map<String, Set<PhysicalTableDefinition>> tableDefinitions =
+            new HashMap<>();
+    private ConfigLoader configLoader;
```

Comment:
These variable names end up being confusing below because it's not clear that they are maps by table name.